connectors/sources/sharepoint_server.py

```diff
@@ -515,8 +657,246 @@ def get_default_configuration(cls):
+                "type": "bool",
+                "value": True,
+            },
+            "fetch_unique_list_item_permissions": {
+                "depends_on": [{"field": "use_document_level_security", "value": True}],
+                "display": "toggle",
+                "label": "Fetch unique list item permissions",
+                "order": 11,
+                "tooltip": "Enable this option to fetch unique list item permissions. This setting can increase sync time. If this setting is disabled a list item will inherit permissions from its parent site.",
+                "type": "bool",
+                "value": True,
+            },
         }
 
+    def _dls_enabled(self):
+        if (
+            self._features is None
+            or not self._features.document_level_security_enabled()
+        ):
+            return False
+
+        return self.configuration["use_document_level_security"]
+
+    def _decorate_with_access_control(self, document, access_control):
+        if self._dls_enabled():
+            document[ACCESS_CONTROL] = list(
+                set(document.get(ACCESS_CONTROL, []) + access_control)
+            )
+
+        return document
+
+    def access_control_query(self, access_control):
+        return es_access_control_query(access_control)
+
+    async def _user_access_control_doc(self, user):
+        login_name = user.get("LoginName")
+        prefixed_mail = _prefix_email(user.get("Email"))
+        prefixed_title = _prefix_user(user.get("Title"))
+        prefixed_user_id = _prefix_user_id(user.get("Id"))
+        prefixed_login_name = _prefix_login_name(login_name)
+
+        access_control = [
+            prefixed_mail,
+            prefixed_title,
+            prefixed_user_id,
+            prefixed_login_name,
+        ]
+
+        return {
+            "_id": login_name,
+            "identity": {
+                "email": prefixed_mail,
+                "title": prefixed_title,
+                "user_id": prefixed_user_id,
+                "login_name": prefixed_login_name,
+            },
+            "created_at": iso_utc(),
+        } | self.access_control_query(access_control)
+
+    async def _access_control_for_member(self, member):
+        principal_type = member.get("PrincipalType")
+        is_group = principal_type != 1 if principal_type else False
+        user_access_control = []
+
+        if is_group:
+            group_name = member.get("Title")
+            self._logger.debug(f"Detected group '{group_name}'.")
+            group_id = _get_login_name(member.get("LoginName"))
+
+            if group_id:
+                user_access_control.append(_prefix_group(group_id))
+            if group_name:
+                user_access_control.append(_prefix_group_name(group_name))
+
+        else:
+            login_name = _get_login_name(member.get("LoginName"))
+            email = member.get("Email")
+            user_access_control.append(_prefix_user_id(member.get("Id")))
```

Comment:
Add debug log to complement one above. This has the benefit of logging when email or login_name are missing

```suggestion
            login_name = _get_login_name(member.get("LoginName"))
            email = member.get("Email")
            id = member.get("Id")
            self._logger.debug(f"Detected member '{id}' with login '{login_name}:{email}' and principal type '{principal_type}', processing as individual.")
            
            user_access_control.append(_prefix_user_id(id))
```